static/assets/contribute-templates/how-to-template.md

```diff
@@ -1,36 +1,631 @@
+---
+**WARNING**
 
-## Before you begin [required if there are prerequisites]
+This guide should be viewed as a **last resort**, as it is a **risky** approach to this problem and can ruin your chain upgrades if something is done incorrectly. Perform this at your own risk.
 
-Replace this template content with an introductory sentence and list of prerequisites similar to the following.
+---
 
-Before you begin, verify the following requirements:
+## Use cases
 
-- What should someone **have** before reading this article?
+This is particularly important if you've accidentally shrunk your block length too much and need to bring your chain back to an upgrade-able state. 
 
-- What should someone **know** before reading this article?
+## Before you begin 
 
-- What should someone **do** before reading this article?
+Before you begin, you should have some knowledge of Forkless Runtime upgrades, you can find a guide on this here: https://docs.substrate.io/tutorials/v3/forkless-upgrades/.
 
-## Steps [use a more descriptive heading as needed]
 
-Replace this paragraph with at least one introductory sentence and provide a stem clause that leads into the first step. For example, use the following this pattern.
+## How do I recover my chain?
 
-To submit an article:
+We will use our experience at Bittensor as an example for this how-to guide. What we had done was accidentally shrink our block length to 65KB in an effort to over-optimize. This made it impossible to run forkless chain upgrades as the WASM files we were uploading were far too large for the blocks at 300KB. As a result, we had to "thread the needle". That is, we created a very small runtime (64.8KB) that consisted of only the ability to perform a chain upgrade with a larger block length, then followed it with another upgrade that restored the code back to its working state. 
 
-1. Open ...
+It's important to prepare both runtimes and push them consecutively. First prepare the normal runtime you are trying to upgrade with and save it aside for use in step 10. Now, let's create this minimalistic runtime. It must be smaller than your present block length. In our case, the block length was 65KB. All of the following changes occur in `runtime/src/lib.rs`.
 
-## Examples [required - link to at least one example]
 
-Replace this paragraph with an introductory sentence followed by a link to at least one working example that corresponds with the information covered in this guide.
+1. Remove any and all custom code you added to your runtime. This means removing the implementation from `construct_runtime!`, all associated `parameter_types!`, and all implementations for runtime.
 
-## Related resources [optional - add links as needed]
+2. Comment out some unneeded configs and functions. Things like `native_version()`, `frame_benchmarking`, and `pallet_transaction_payment_rpc_runtime_api` can all be removed. 
```

Comment:
`frame_benchmarking` is not part of the final wasm, unless if you are mistakenly enabling its feature. Under correct circumstances, removing it should not affect the wasm size at all.

Reply:
Interesting, I'll add this in.